generators/reference_chunking/newline_splitter/__init__.py

```diff
@@ -0,0 +1,22 @@
+from pydantic import BaseModel
+
+INPUT_EXAMPLE = {
+    "text": """This is the first line.
+    And this is the second line.
+    Here's a third one, too.
+    """,
+}
+
+class NewlineSplitterModel(BaseModel):
+    text: str
+
+    class Config:
+        schema_extra = {"example": INPUT_EXAMPLE}
+
+
+def newline_splitter(req: NewlineSplitterModel):
+    """Splits a text by newline characters"""
+    splits = req.text.strip().split("\n")
+    return {"splitted_text" : [val for val in splits if len(val) > 0]}
```

Comment:
For a text 
```
Hello
 
abc
```

this would result in three chunks (since i added a space in the second line).

Maybe we shouldn't strip on the full text but rather on the chunks as well:
```suggestion
    splits = req.text.split("\n")
    return {"splitted_text" : [val for val in splits if len(val.strip()) > 0]}
```


This would result in different results if someone started with spaces and want to preserve them. For the embedding list usecase i think we want only the text parts so i'd go with the suggestion